front_end/sdk/DartSupport.js

```diff
@@ -5,6 +5,42 @@
 // that happen?
 Dart = {};
 
+/// A handler for dealing with system events.
+Dart._NotificationHandler = class {
+
+  static register() {
+    SDK.targetManager.addModelListener(
+      SDK.DebuggerModel, SDK.DebuggerModel.Events.DebuggerPaused, this._debuggerPaused);
+  }
+
+  static async _isInDartContext() {
+    // TODO(alanknight): Reduce code duplication with $dartExpressionFor.
+    const executionContext = UI.context.flavor(SDK.ExecutionContext);
+    // We can't use selectedCallFrame() because it hasn't been set yet.
+    const selectedFrame = executionContext.debuggerModel.callFrames[0];
+    // We're not stopped at a breakpoint, treat it as JS.
+    if (!selectedFrame) return false;
```

Comment:
nit: why shouldn't the default be to treat it as dart? This tool is for Dart users.

Reply:
It should, and there's a bug for that, but we need to do it differently if we don't have a frame to get the information from, and that isn't done yet.

Reply:
makes sense. maybe add a TODO to make the default dart in the future?